test/gtest/uct/test_amo.cc

```diff
@@ -21,6 +21,13 @@ void uct_amo_test::init() {
     entity *receiver = uct_test::create_entity(0);
     m_entities.push_back(receiver);
 
+    try {
+        check_skip_test();
+    } catch (...) {
+        cleanup();
```

Comment:
entity destructor is called from `uct_test::cleanup`, but `uct_test::cleanup` isn't called, because
```
void test_base::SetUpProxy() {
    ...
    try {
        check_skip_test(); 
        init();
        m_initialized = true;
        m_state = RUNNING;
    } catch (test_skip_exception& e) {
        skipped(e);
    } ...
}
```
```
void test_base::TearDownProxy() {
    ...
    if (m_initialized) {
        cleanup();
    }
}
```

Reply:
if init() was not completed, cleanup() should not be called - that's ok
but maybe we can realy on the destructors here to clean up instead of catch (...) ?

Reply:
ah, got it
will do